services/manifest-repo-export-service/pkg/repository/transformer.go

```diff
@@ -2139,10 +2125,76 @@ func (c *DeleteAAEnvironmentConfig) GetEslVersion() db.TransformerID {
 }
 
 func (c *DeleteAAEnvironmentConfig) Transform(
-	_ context.Context,
-	_ *State,
-	_ TransformerContext,
+	ctx context.Context,
+	state *State,
+	tCtx TransformerContext,
 	_ *sql.Tx,
 ) (string, error) {
-	return GetNoOpMessage(c)
+	fs := state.Filesystem
+
+	envDir := fs.Join("environments", string(c.Environment))
+	if err := fs.MkdirAll(envDir, 0777); err != nil {
+		return "", err
+	}
+
+	configFile := fs.Join(envDir, "config.json")
+
+	data, err := util.ReadFile(fs, configFile)
+	if err != nil {
+		return "", fmt.Errorf("error opening file: %w", err)
+	}
+	var envConfig config.EnvironmentConfig
+
+	err = json.Unmarshal(data, &envConfig)
+	if err != nil {
+		return "", err
+	}
+
+	argoCdAppFile := fs.Join("argocd", string(argocd.V1Alpha1), fmt.Sprintf("%s.yaml", *envConfig.ArgoCdConfigs.CommonEnvPrefix+"-"+string(c.Environment)+"-"+string(c.ConcreteEnvironmentName)))
```

Comment:
The same file is also written in "afterTransform". Is this line really necessary?
We shouldn't write the same thing in two different places.

Reply:
At least this line that is just putting together the path of the file, should be extracted into its own function. (So that it's a bit more obvious that other code is calling this too).